app/src/main/java/com/group80/uoftinder/RecommendationInteractor.java

```diff
@@ -0,0 +1,82 @@
+package com.group80.uoftinder;
+
+import com.group80.uoftinder.entities.User;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+public class RecommendationInteractor {
+
+//    private User currentUser;
+    private List<User> compatibilityList;
+//    private List<String> compatibilityList;
+
+    public RecommendationInteractor() {
+
+    }
+
+    /**
+     * Mutates the instance variable compatibilityList based on criteria given by
+     * filters parameter and minimum/maximum age values. Also updates compatibilityList
+     * of currentUser, beyond just mutating the instance variable compatibilityList.
+     *
+     * @param filters   A list of sets of integers that represents the wanted criteria.
+     *                  Each set in the list corresponds to a filtering question.
+     *                  For academic users, filters.size() should be 3 (program of study,
+     *                  year of study, and campus). Each integer represents a selected choice.
+     *                  {0, 3, 5} means the first, fourth, and sixth choices were checked.
+     * @param minAge    The minimum age, inclusive
+     * @param maxAge    The maximum age, inclusive
+     */
+    public void filterCompatibilityList(List<Set<Integer>> filters, int minAge, int maxAge) {
+        for(int j = compatibilityList.size() - 1; j >= 0; j--) {
+            User user = compatibilityList.get(j);
+            if(user.getAge() < minAge || user.getAge() > maxAge) {
+                compatibilityList.remove(j);
+                continue;  // go to the next user
+            }
+            // answers are formatted in the same way as filters, explained above
+            List<Set<Integer>> answers = user.getAnswers();
+            for(int i = 0; i < answers.size(); i++) {
+                Set<Integer> currentFilter = filters.get(i);
```

Comment:
This code operates on the assumption that the number of sets within answers and filters will always be the same (`filters.size() == answers.size()`), which should be true since the filters are created based on the exact questions that users have to answer to build their profile.

Reply:
Okay, I see. Maybe you could specifically comment that in your code as an invariant. I'm a bit concerned that as the number of questions gets larger (for instance, if we include 10 questions like we were planning to at one point), there may not be enough space on screen to display every option for each question. Have you considered what will happen at that stage?

Reply:
Very valid concern, I considered that as well. My solution was with the UI where I implemented a scrollable view so that when the number of questions increase by a lot, the filters would still fit "on" the screen due to its scrollable nature. Just added more comments about the invariant in my most recent commit!